doc/plutus-core-spec/flat-serialisation.tex

```diff
@@ -195,16 +195,17 @@ \subsection{Arrays}
 
 \subsection{Natural numbers}
 We encode natural numbers by splitting their binary representations into
-sequences of 7-bit blocks, then emitting these as a list with the \textbf{least
-  significant block first}:
+sequences of 7-bit blocks, then emitting all but last chunks as a list with
+the \textbf{least significant block first}. The last chunk(most significant chunk)
+will be encoded as regular 7-bit block and appended right after the list of chunks.
 
 $$
-\E_{\N} (s, \sum_{i=0}^{n-1}k_i2^{7i}) = \Elist_7(s, [k_0, \ldots, k_{n-1}])
+\E_{\N} (s, \sum_{i=0}^{n-1}k_i2^{7i}) = \E_7(\Elist_7(s, [k_0, \ldots, k_{n-2}]), k_{n-1})
 $$
 \noindent(where $k_i \in \Z$ and $0 \leq k_i \leq 127$).
 \noindent The decoder is
 $$
-\D_{\N}(s) = (s', \sum_{i=0}^{n-1}k_i2^{7i}) \quad \text{if $\Dlist_7(s) = (s', [k_0, \ldots, k_{n-1}])$}.
+\D_{\N}(s) = (s'', \sum_{i=0}^{n-1}k_i2^{7i}) \quad \text{if $\Dlist_7(s) = (s', [k_0, \ldots, k_{n-2}])$ and $\D_7(s') = (s'', k_{n-1})$}.
```

Comment:
```suggestion
\D_{\N}(s) = (s'', \sum_{i=0}^{n}k_i2^{7i}) \quad \text{if $\Dlist_7(s) = (s', [k_0, \ldots, k_{n-1}])$ and $\D_7(s') = (s'', k_{n})$}.
```

Reply:
Now I'm wondering what happens if there's only one block, say if your number is 17.  In that case you get

D_list(s) = (s', []) and D_7(s') = (s'', 17)

so you'll get [k_0, ..., k_{n-1}] = [] and k_n = 7.  I suppose we could claim that [k_0, ..., k_{n-1}] = [] forces n=0 (but why not n=-5 or something, which would also give you the empty list?), and then the result is k_{0}7^{0}0=7.  Maybe we need a case to deal specially with the empty list. We could also bypass the list decoder altogether and write down the full details of how the decoder for N works, but that might need qutie a lot of extra notation.

Reply:
I make explicit case for when there is only one block to make this clearer